### ISSUE TITLE
require-presentational-children requires role="presentation" on component invocations

### ISSUE THREAD

**Issue by runspired** [ember-template-lint/ember-template-lint#2334]
I'm not sure if this is intentional, or a bug, but it's an artifact of this line: https://github.com/ember-template-lint/ember-template-lint/blob/523d93cddb1953fb36111da33603270598da2e52/lib/rules/require-presentational-children.js#L96

I suspect the intent was to future-proof the list of semantic elements, if so then this check needs to filter out component invocations such as `<@aComponent />` `<AComponent />` and `<:aComponent />`

of course, maybe the rule is also operating as intended. Since the contents of components are rendered into the context of the role they too should be presentational-only; however, this would in practice necessitate also linting the contents of each component found to be invoked within such a role which we do not yet do, and in practice (ime) more components do not have a top-level tag with splattributes than do, so potentially adding attributes where none will be utilized feels like bad form (but maybe isn't? I don't have a well formed/researched opinion on this).

cc @MelSumner

**Comment by lifeart** [comment]
I like idea to ignore names, looking like components (has capitalized letters, has dot in name, has ':' as first character)

**Comment by kangax** [comment]
I ran into the same issue with custom components in our codebase. This rule is reporting false positives (more like: positives or negatives depending on if custom component adds role="presentation" which the rule obviously has no way of knowing about):

For example:
```
<ArtdecoDropdown$ArtdecoDropdownItem
  role="button"
  ...
>
  <SmpServicePages$ServicesPageTopCard::HeaderOverflowAction
    @overflowAction={{overflowAction}}
    @headerOverflowActionTextId={{actionTextId}}
  />
</ArtdecoDropdown$ArtdecoDropdownItem>
```

We considered just adding `role=presentation` to all of these custom components but that feels like a hacky workaround while fixing the rule upstream would be the right way.

Looking at [no-nested-interactive](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-nested-interactive.md) there's `ignoredTags` and `additionalInteractiveTags` allowing for customization.

What if we add `ignoreCustomTags: true | false | Array<string>` as an option for `require-presentational-children`? 

I can make a PR for this if this is an acceptable solution.

**Comment by MelSumner** [comment]
I think we want the outcome to be "hey anything you put inside of a button element is going to be considered presentational by browsers"

Like, that's what the browsers do _now_, but how do we communicate this to developers? 

I agree that if the rule is requiring `role="presentation"` then that feels bad; `div` and `span` elements have an implicit role of none so they shouldn't be triggering this rule. 

> What if we add ignoreCustomTags: true | false | Array<string> as an option for require-presentational-children? 

I think that would be okay, @kangax .
